src/main/initGridCellsMod.F90

```diff
@@ -244,12 +245,25 @@ subroutine set_landunit_veg_compete (ltype, gi, li, ci, pi)
        call add_column(ci=ci, li=li, ctype=1, wtlunit=1.0_r8)
        ncols_added = ncols_added + 1
 
-       do m = natpft_lb,natpft_ub
-          if (natveg_patch_exists(gi, m)) then
-             call add_patch(pi=pi, ci=ci, ptype=m, wtcol=wt_nat_patch(gi,m))
+       if(use_fates .and. .not.use_fates_sp)then
```

Comment:
I tried this but ran into a snag.  If FATES is run in a mode that has fixed biogeography and no-competition but not SP, the wt_nat_patch array does need to have meaningful values, but the size of the array will not necessarily match the number of fates patches (defined by maxveg).  So we can't feed wt_nat_patch into this routine in all situations.

Reply:
Ahh, OK. It seems like for FATES we should have natpft_ub set to maxveg. I think that's actually reasonable to do because natpft_ub isn't actually used that often. But, haven't it wrong for FATES cases isn't helpful. With this comment I was thinking that you had already done that in clm_varpar, but you are right that isn't done now. I do think it should be though.

Reply:
@ekluzek and I discussed this in depth.  We realized that since the number of patches allocated on the natural veg LU, is not necessarily equal to the size of the wt_nat_patch array anymore, it is impossible to have the wt_nat_patch drive the patch area weights at this step for FATES.  This is because  there is not a 1 to 1 mapping of pfts in the surface file, to pfts in a fates biogeography mode simulation.

Reply:
however, we did agree rework the code so that the array bounds natpft_lb and natpft_ub will retain their meaning in a FATES simulation.  In my first version of this refactor, I was assuming natpft_lb and natpft_ub were the array bounds for pfts in the surface file (with fates on).  now they are assumed to be the array bounds for the allocated patches on the nat LU, which is more consistent with its use in non-fates.